api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -664,14 +664,58 @@ public static String capitalizeFirstCharacterOnly(String text) {
     return StringUtils.capitalize(text.toLowerCase());
   }
 
+  private static String generateSqlWithEnvironmentVariables(
+      String query, KernelTypeEnum kernelTypeEnum) {
+    switch (kernelTypeEnum) {
+      case PYTHON:
+        return query.replaceAll(
+            "\\$\\{projectId}.\\$\\{dataSetId}", "\"\"\" + os.environ[\"WORKSPACE_CDR\"] + \"\"\"");
```

Comment:
Did you look at str_glue as Nicole suggested? I think it would be more readable to declare variables then substitute them into the templated SQL as is done in the snippets. Note: it would require setting up an additional package

Reply:
+1 . I get scared seeing system calls in the middle of a complex string operation. We need to have the cleanest, earliest error code we can if one of those values is missing or malformed.

Also, I'll give you a dollar for every string or regex replacement we can kill in this class. 😉 I believe we can build almost everything up in one direction, and that's much easier to reason about and test. Later PR of course.

Reply:
Where is the str_glue comment? I didn't see that in the email I got.

I do hate the amount of string replacement we're doing, but we would need to go into the cohort builder sql builder to change it too much. (Unfortunately we're using the same SQL building code that they use.)

Reply:
Sorry I'm not sure if she explicitly suggested this, I was more thinking you were basing the approach on the code she had already written for the snippets menu. Here's an example of str_glue: https://github.com/all-of-us/workbench-snippets/blob/f6e714328ba8c0d3ff11387944fccb0df5ba7ab5/sql-snippets/snippets_setup.R#L30 ; see more complicated variants by inserting from the snippets SQL menu. The relevant thing here is that it lets you interpolate variables into a string.